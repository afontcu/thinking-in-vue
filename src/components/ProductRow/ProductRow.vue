<template>
  <tr class="c-product-row">
    <td class="c-product-row__name">
      {{ name }}
    </td>
    <td class="c-product-row__price">
      {{ price }} â‚¬
      <span v-if="offer" class="c-product-row__offer">offer!</span>
    </td>
  </tr>
</template>

<script>
export default {
  name: 'ProductRow',
  props: {
    name: {
      type: String,
      required: true
    },
    price: {
      type: Number,
      required: true
    },
    offer: {
      type: Boolean,
      required: true
    }
  }
}
</script>

<style>
td {
  position: relative;
}
.c-product-row__name {
  text-align: left;
}
.c-product-row__price {
  text-align: right;
}
.c-product-row__offer {
  position: absolute;
  left: 1rem;
  top: 23px;
  color: #8ea631;
  font-style: italic;
  text-transform: uppercase;
  font-size: 0.75rem;
  font-weight: bold;
  letter-spacing: 0.2px;
}
</style>
