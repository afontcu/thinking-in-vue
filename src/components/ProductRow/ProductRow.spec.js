import ProductRow from './index.js'
import { mount } from '@vue/test-utils'

const propsData = {
  name: 'name',
  price: 12,
  offer: false
}

describe('ProductRow.vue', () => {
  it('is a component', () => {
    const wrapper = mount(ProductRow, { propsData })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders an snapshot', () => {
    const wrapper = mount(ProductRow, { propsData })
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
