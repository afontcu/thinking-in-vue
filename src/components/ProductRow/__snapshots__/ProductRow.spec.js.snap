// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow.vue renders an snapshot: default 1`] = `
<tr>
    <td>
        <!---->
        name
    </td>
(??)
</tr>
`;
